celery/worker/consumer/connection.py

```diff
@@ -18,14 +18,19 @@ def __init__(self, c, **kwargs):
         super().__init__(c, **kwargs)
 
     def start(self, c):
-        c.connection = c.connect()
-        info('Connected to %s', c.connection.as_uri())
+        try:
+            c.connection = c.connect()
+            info('Connected to %s', c.connection.as_uri())
+        except Exception as e:
+            logger.error('Failed to connect: %s', str(e))
```

Comment:
Better messaging will help during debugging

Reply:
At this point, we need to rely on the exception message passed by the handler.